internal/runners/swtch/switch.go

```diff
@@ -0,0 +1,133 @@
+package swtch
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/installation/storage"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runbits"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+type Switch struct {
+	auth      *authentication.Auth
+	out       output.Outputer
+	project   *project.Project
+	analytics analytics.Dispatcher
+	svcModel  *model.SvcModel
+}
+
+type SwitchParams struct {
+	Identifier string
+}
+
+type primeable interface {
+	primer.Auther
+	primer.Outputer
+	primer.Projecter
+	primer.Configurer
+	primer.Analyticer
+	primer.SvcModeler
+}
+
+type identifier interface {
+	CommitID() strfmt.UUID
+	Type() string
```

Comment:
It seems you are only using this for localization, so I recommend naming it Locale() and passing it through the locale package.

Otherwise this should probably be either more strongly typed or have constants to compare the value to as we don't want to be doing manual string matching.